### PR TITLE
#1872 Document brick. Card component + Tests.

### PR DESCRIPTION
**Pull request by BALEHOK**
Relates to #1872

### CODE REVIEW

File: src/blocks/renderers/documentView/documentView.test.tsx [hunk 1]
```diff
@@ -15,14 +15,59 @@
...
+  (headerType: string) => {
+    const config = {
+      type: headerType,
+      config: {
+        title: "Test Header",
+        className: "test-class",
+      },
+    };
+    const rendered = renderDocument(config);
+    expect(rendered.asFragment()).toMatchSnapshot();
```
Comment (twschiller) [inline src/blocks/renderers/documentView/documentView.test.tsx:40]: I'm OK including a snapshot tests, but we also have to have more meaningful assertions.

In this case, looking for an h1, h2, h3, etc. that contains the text

**Review by twschiller — CHANGES_REQUESTED** [review]
See comment on not relying solely on snapshots

File: src/blocks/renderers/documentView/documentView.test.tsx [hunk 2]
```diff
@@ -49,10 +124,54 @@ test("basic test for dev purposes", () => {
...
+        heading: "Test Heading of Card",
+        body: "Test body of card",
+      },
+    };
+    const rendered = renderDocument(config);
+    expect(rendered.asFragment()).toMatchSnapshot();
+  });
 
-  expect(container).not.toBeNull();
+  // FIXME This one fails to render the pipeline, see the snapshot
```
Comment (twschiller) [inline src/blocks/renderers/documentView/documentView.test.tsx:161]: This is a great example of why snapshots aren't sufficient. Just from looking at the test you don't know whether or not the brick is meant to be in a loading state/error state/or successfully showing the content

You'd have to also open up the snapshot

Comment (twschiller) [reply src/blocks/renderers/documentView/documentView.test.tsx]: @BALEHOK I've added an example of more detailed assertions here and in the pipeline test

**Review by twschiller — APPROVED** [review]
@BALEHOK I'm merging this in, but see comments about not relying solely on snapshots in Jest tests

File: src/blocks/renderers/documentView/documentView.test.tsx (hunk 2 above)
Comment (BALEHOK) [reply src/blocks/renderers/documentView/documentView.test.tsx]: > You'd have to also open up the snapshot
This is correct. You always have to look, read, and understand the snapshot with snapshot testing.

With the tiny elements like `h1` or `p` we want to assert on pretty much everything: tag, attributes, content. One snapshot can replace multiple assertions.

If we add explicit assertions on all of those properties, then the snapshot itself is not needed.

PR: https://github.com/pixiebrix/pixiebrix-extension/pull/1973

**Pull request merged**